### PR TITLE
Change shard ID type from int to int32

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Unify the shard ID type to int32
* Remove unused admin Kafka resend command since we already have auto resend & Kafka is going to be deprecated

**Why?**
This PR provides better test coverage and code management.

**How did you test it?**
Run tests.

**Potential risks**
Minor business logic code change.

### CODE REVIEW

File: service/history/replicationTaskProcessor.go [hunk 1]
```diff
@@ -431,7 +431,7 @@ func (p *ReplicationTaskProcessorImpl) putReplicationTaskToDLQ(replicationTask *
 	p.metricsClient.Scope(
 		metrics.ReplicationDLQStatsScope,
 		metrics.TargetClusterTag(p.sourceCluster),
-		metrics.InstanceTag(strconv.Itoa(p.shard.GetShardID())),
+		metrics.InstanceTag(strconv.Itoa(int(p.shard.GetShardID()))),
```
Comment (sergeybykov) [inline service/history/replicationTaskProcessor.go:434]: Use instead `convert.Int32ToString` that was added in #762 for this purpose.

File: service/history/shardContext.go [hunk 2]
```diff
@@ -286,7 +286,7 @@ func (s *shardContextImpl) UpdateReplicatorDLQAckLevel(
 	s.GetMetricsClient().Scope(
 		metrics.ReplicationDLQStatsScope,
 		metrics.TargetClusterTag(sourceCluster),
-		metrics.InstanceTag(strconv.Itoa(s.shardID)),
+		metrics.InstanceTag(strconv.Itoa(int(s.shardID))),
```
Comment (sergeybykov) [inline service/history/shardContext.go:289]: Use instead convert.Int32ToString that was added in #762 for this purpose.

File: common/archiver/filestore/queryParser.go [hunk 3]
```diff
@@ -251,17 +251,17 @@ func convertToTime(timeStr string) (time.Time, error) {
 func convertStatusStr(statusStr string) (enumspb.WorkflowExecutionStatus, error) {
 	statusStr = strings.ToLower(strings.TrimSpace(statusStr))
 	switch statusStr {
-	case "completed", strconv.Itoa(int(enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED)):
+	case "completed", convert.Int32ToString(int32(enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED)):
```
Comment (alexshtin) [inline common/archiver/filestore/queryParser.go:254]: These are not shardIDs. Actually, can you change it to:
```suggestion
	case "completed", enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED.String():
```

File: common/cache/namespaceCache.go [hunk 4]
```diff
@@ -119,8 +119,8 @@ type (
 		refreshLock sync.Mutex
 
 		callbackLock     sync.Mutex
-		prepareCallbacks map[int]PrepareCallbackFn
-		callbacks        map[int]CallbackFn
+		prepareCallbacks map[int32]PrepareCallbackFn
```
Comment (alexshtin) [inline common/cache/namespaceCache.go:122]: Is it shardID?

File: service/history/handler.go [hunk 5]
```diff
@@ -1366,7 +1366,7 @@ func (h *Handler) SyncShardStatus(ctx context.Context, request *historyservice.S
 	}
 
 	// shard ID is already provided in the request
-	engine, err := h.controller.getEngineForShard(int(request.GetShardId()))
+	engine, err := h.controller.getEngineForShard(int32(request.GetShardId()))
```
Comment (alexshtin) [inline service/history/handler.go:1369]: You can change it in protos too. Just remove `int64` field completely and add new `int32 ShardId`. Let's do it as separate PR.

File: tools/cli/admin.go [hunk 6]
```diff
@@ -556,74 +556,6 @@ clusters:
 				AdminListDLQ(c)
 			},
 		},
-		{
```
Comment (alexshtin) [inline tools/cli/admin.go:559]: This is definitely shouldn't be here.

**Review by alexshtin — COMMENTED** [review]
I like this change!

File: common/cache/namespaceCache.go (hunk 4 above)
Comment (wxing1292) [reply common/cache/namespaceCache.go:122]: yes, ref:
```go
c.prepareCallbacks[shard] = prepareCallback
```

File: service/history/handler.go (hunk 5 above)
Comment (wxing1292) [reply service/history/handler.go:1369]: SyncShardStatus should be done as part of NDC readiness. I will handle it in another PR :)

**Review by alexshtin — APPROVED** [review]

File: common/archiver/filestore/queryParser.go (hunk 3 above)
Comment (wxing1292) [reply common/archiver/filestore/queryParser.go:254]: actually the intention here is to get the integer value for query parsing, string function will return the actual enum name, like `Completed`
